app/src/main/java/ip/team13/petowner/ui/leaderboard/LeaderboardViewModel.kt

```diff
@@ -1,29 +1,38 @@
 package ip.team13.petowner.ui.leaderboard
 
 import androidx.databinding.Bindable
-import androidx.databinding.ObservableField
 import ip.team13.petowner.core.BaseViewModel
 import ip.team13.petowner.data.domain.LeaderboardEntry
 import ip.team13.petowner.data.domain.LeaderboardType
 import ip.team13.petowner.data.repository.LeaderboardRepository
+import androidx.databinding.library.baseAdapters.BR
 
 class LeaderboardViewModel(
     private val leaderboardRepository: LeaderboardRepository
 ) : BaseViewModel() {
 
-    val leaderboardType = ObservableField(LeaderboardType.FREE)
-
-    val person1 = ObservableField<LeaderboardEntry>()
-    val person2 = ObservableField<LeaderboardEntry>()
-    val person3 = ObservableField<LeaderboardEntry>()
+    @get:Bindable
+    var leaderboardType: LeaderboardType = LeaderboardType.FREE
+        set(value) {
+            field = value
+            notifyChange()
```

Comment:
You don't need the `notifyChange`. All the `@Bindable` fields automatically notify the UI on any update

Reply:
Does it notify all the fields? When the leaderboard type is changed, also items and person{1,2,3} will be changed.

Reply:
Ah ok. No it doesn't